skilld-labs/zen
# Adding Webpack v3 and Babel for ES6 Support

Added gulp task `scripts` and webpack is handled with this task. If everything is ok I will update README to reflect the changes.

Steps to test:

1- install fresh drupal
2- clone repo (github.com/greteltrost/zen) and checkout to 'webpack' branch
3- install new theme from starterkit and enable it
4- test es6 by adding some sample code on js/init.js:

```js
const double = [1,2,3].map((num) => num * 2);
```

5- build js with webpack by running `gulp` or `NODE_ENV=production gulp` for minified version (or only javascript with `gulp scripts`)
6- check new code, should be transpiled to something that all browsers should be capable to run on /dist/app.js

```js
"use strict";
var double = [1, 2, 3].map(function (num) {
  return num * 2;
});
```

More info about webpack on [https://webpack.js.org/concepts/](Webpack)
